tztrout/data.py

```diff
@@ -90,8 +97,27 @@ def deduplicate(item):
     return deduplicate
 
 
+def generate_us_zipcode_namedtuples():
+    """Generate a list of namedtuples for US zipcodes from the list of
+    dictionaries in data/us_zipcode_data.json
+    """
+    if not os.path.exists(US_ZIPCODE_DATA_PATH):
+        raise ValueError(
+            'US Zipcode data missing. Run regenerate_data.py first'
+        )
+    us_zipcode_tuples = []
+    with open(US_ZIPCODE_DATA_PATH, 'r') as file:
+        data = json.loads(file.read())
+    for zip in data:
+        us_zipcode_tuples.append(Zipcode(**zip))
```

Comment:
From usages below, we don't need to collect the entire list in memory. Generate them lazily with `yield Zipcode(...)`

Reply:
The initial switch is done in 79b5bd4. 

This one wound up being a little tricky because when I used yield and made this method a generator, `_progressbar` stopped working because you cannot `len` a generator. So, I made my own version of a progressbar for generators in c25ce05